src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    if (`${item[key]}` in groupedItems) {
```

Comment:
better to create separate variable for item[key], you use it 3 times

Reply:
and interpolation looks quite strange here, think of a way to remove it

Reply:
Done!